scripts/pythondistdeps.py

```diff
@@ -11,21 +11,79 @@
 # RPM python dependency generator, using .egg-info/.egg-link/.dist-info data
 #
 
-# Please know:
-# - Notes from an attempted rewrite from pkg_resources to importlib.metadata in
-#   2020 can be found in the message of the commit that added this line.
-
 from __future__ import print_function
 import argparse
-from os.path import basename, dirname, isdir, sep
-from sys import argv, stdin, version
 from distutils.sysconfig import get_python_lib
+from itertools import chain
+from os.path import basename, dirname, isdir, sep
+from packaging.requirements import Requirement
+from packaging.version import parse
+import re
+from sys import argv, stdin
 from warnings import warn
 
+try:
+    from importlib.metadata import PathDistribution
+except ImportError:
+    from importlib_metadata import PathDistribution
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+
+class Req(Requirement):
+    @property
+    def key(self):
+        return self.name.lower()
+
+
+class Distribution(PathDistribution):
+    def __init__(self, path):
+        super(Distribution, self).__init__(Path(path))
+        self.name = self.metadata['Name']
+        self.key = re.sub('[^A-Za-z0-9.]+', '-', self.name).lower()
+        self.requirements = [Req(r) for r in self.requires or []]
+        self.extras = [
+            v for k, v in self.metadata.items() if k == 'Provides-Extra']
+        self.py_version = self._parse_py_version(path)
+
+    def _parse_py_version(self, path):
+        # Try to parse the Python version from the path the metadata
+        # resides at (e.g. /usr/lib/pythonX.Y/site-packages/...)
+        if isdir(path):
+            path = dirname(path)
+        res = re.search(r"/python(?P<pyver>\d+\.\d+)/", path)
```

Comment:
This breaks detection of Python version from the name of the egg-info file. And I think you misunderstood me, I wasn't doubting the usefulness of the variable, I was asking why do you do `dirname` at all.
```suggestion
        res = re.search(r"/python(?P<pyver>\d+\.\d+)/", path)
```